### PR TITLE
Fix/wallet-connect-connections

### PR DESCRIPTION
**Pull request by LukeSamkharadze**

### CODE REVIEW

File: src/widgets/WalletModal/WalletCard.tsx [hunk 1]
```diff
@@ -20,10 +21,22 @@ const WalletCard: React.FC<Props> = ({ login, walletConfig, onDismiss, mb }) =>
...
-        window.localStorage.setItem(connectorLocalStorageKey, walletConfig.connectorId);
-        onDismiss();
+        // TW point to WC on desktop
+        if (title === "TrustWallet" && walletConnectConfig && isDesktop) {
+          login(walletConnectConfig.connectorId);
+          localStorage?.setItem(walletLocalStorageKey, walletConnectConfig.title);
+          localStorage?.setItem(connectorLocalStorageKey, walletConnectConfig.connectorId);
+          onDismiss();
+          return;
+        }
```
Comment (erwin-huang) [inline src/widgets/WalletModal/WalletCard.tsx:31]: I think we can simplify this part, we can remove `line 24` to `line 31`

for `line 32`, we can use this
`if (!window.ethereum && walletConfig.redirectUrl && walletConfig.connectorId !== ConnectorNames.WalletConnect) {`

and in `src/widgets/WalletModal/config.tsx`
we can update TrustWallet Object to this
```
{
    title: "TrustWallet",
    icon: TrustWallet,
    connectorId: isDesktop ? ConnectorNames.WalletConnect :  ConnectorNames.Injected,
    redirectUrl: `https://link.trustwallet.com/open_url?coin_id=20000714&url=https://summitswap.finance/`
  },
  ```

File: src/widgets/WalletModal/ConnectModal.tsx [hunk 2]
```diff
@@ -20,39 +20,34 @@ const HelpLink = styled(Link)`
...
-            walletConfig={entry}
-            onDismiss={onDismiss}
-            mb={index < config.length - 1 ? "8px" : "0"}
-          />
-        </a>
-      ) : (
+const ConnectModal: React.FC<Props> = ({ login, onDismiss = () => null }) => {
+  const walletsToShow = window.ethereum?.isTrust
+    ? config.filter((wallet) => wallet.title !== "WalletConnect")
+    : config;
```
Comment (erwin-huang) [inline src/widgets/WalletModal/ConnectModal.tsx:26]: Why do we need to hide `WalletConnect` when it's on trust wallet browser?

Comment (LukeSamkharadze) [reply src/widgets/WalletModal/ConnectModal.tsx:26]: Bcs they can just click `Trust wallet` option adding Wallet connect as well is just confusing

File: src/widgets/WalletModal/WalletCard.tsx (hunk 1 above)
Comment (LukeSamkharadze) [reply src/widgets/WalletModal/WalletCard.tsx:31]: This method was copied from pancakeswap, but yea we can improve it

**Review by erwin-huang — APPROVED** [review]

**Pull request merged**